systemtest/src/test/java/io/strimzi/systemtest/operators/user/UserST.java

```diff
@@ -216,6 +217,42 @@ void testUserWithQuotas(KafkaUser user) {
         });
     }
 
+    @Test
+    void testCreatingUsersWithSecretPrefix() {
+        String clusterName = "second-cluster";
+        String secretPrefix = "top-secret-";
+        String tlsUserName = "encrypted-leopold";
+        String scramShaUserName = "scramed-leopold";
+
+        KafkaResource.kafkaEphemeral(clusterName, 3)
+            .editSpec()
+                .editEntityOperator()
+                    .editUserOperator()
+                        .withNewSecretPrefix(secretPrefix)
+                    .endUserOperator()
+                .endEntityOperator()
+            .endSpec()
+            .done();
+
+        KafkaUserResource.tlsUser(clusterName, tlsUserName).done();
+        KafkaUserResource.scramShaUser(clusterName, scramShaUserName).done();
+
+        Secret tlsSecret = kubeClient().getSecret(secretPrefix + tlsUserName);
+        Secret scramShaSecret = kubeClient().getSecret(secretPrefix + scramShaUserName);
+
+        LOGGER.info("Checking if user secrets with secret prefixes exists");
+        assertNotNull(tlsSecret);
+        assertNotNull(scramShaSecret);
+
+        LOGGER.info("Checking if secrets contains right user names");
+        assertThat(tlsSecret.getMetadata().getOwnerReferences().get(0).getName(), is(tlsUserName));
+        assertThat(scramShaSecret.getMetadata().getOwnerReferences().get(0).getName(), is(scramShaUserName));
+
+        LOGGER.info("Checking if users are created without the secret prefixes");
+        assertNotNull(KafkaUserResource.kafkaUserClient().inNamespace(NAMESPACE).withName(tlsUserName).get());
+        assertNotNull(KafkaUserResource.kafkaUserClient().inNamespace(NAMESPACE).withName(scramShaUserName).get());
```

Comment:
Not sure I see the point of this. The KafkaUser resources are created by you on line 237+238. So you are just checking if you used the right name there. What you really want to check is the username in Kafka and CN of the certificate I guess.